test/test_cpu/test_nvfp4_quant_linear.py

```diff
@@ -0,0 +1,119 @@
+import pytest
+import torch
+
+from auto_round.data_type.nvfp import calculate_gparam
+from auto_round.data_type.utils import get_quant_func
+from auto_round.experimental import qmodules as ar_qmodules
+from auto_round.export.export_to_autoround import AutoRoundFormat
+from auto_round.export.export_to_autoround.qlinear_fp import QuantLinear as _FPLinear
+from auto_round.schemes import PRESET_SCHEMES
+
+QMODULE_MAPPING = {
+    AutoRoundFormat.NVFP4.value: ar_qmodules.NVFP4QuantLinear,
+}
+
+
+@pytest.fixture(autouse=True)
+def fixed_seed():
+    """Ensure deterministic RNG for every test."""
+    seed = 42
+    print("\nSetting fixed random seed for test:", seed)
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
+    yield
+    # (Optional) cleanup or reset after test
+
+
+@pytest.mark.parametrize("scheme", [AutoRoundFormat.NVFP4.value])
+@torch.inference_mode()
+def test_nvfp4_quantlinear_from_original_and_forward(scheme):
+    """
+    Test NVFP4 quantization schemes by creating quantized layers
+    from an original torch.nn.Linear layer and validating their forward pass.
+    """
+
+    # Set random seed for reproducibility
+    torch.manual_seed(42)
+
+    # Define layer dimensions
+    in_features = 64
```

Comment:
there is no feature checker in inference, so you'd better prove it, e.g. changing in_features  to 31 or some values else.
Otherwise, add feature chckers